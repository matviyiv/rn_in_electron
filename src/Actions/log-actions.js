import _ from 'lodash';
import Constants from '../constants';

export function enable(enabled) {
  return {
    type: Constants.LOG.ENABLE,
    data: { enabled },
  };
}

export function logInfo(info, origin, time) {
  return {
    type: Constants.LOG.TYPE.INFO,
    data: {
      message: info,
      origin,
      time: time || new Date().getTime(),
    },
  };
}

export function logWarning(warning, origin, time) {
  return {
    type: Constants.LOG.TYPE.WARNING,
    data: {
      message: warning,
      origin,
      time: time || new Date().getTime(),
    },
  };
}

export function logException(exception, origin, time) {
  /**
   * The reason why we are converting string-errors into Error instances
   * here in the action, is because this will give us a proper stack-trace.
   * If we instead did this in the saga, we would not be able to see where
   * the Error came from, due to the async nature of sagas.
   */
  let _exception = exception;
  if (_.isString(_exception)) {
    _exception = new Error(_exception);
  }
  return {
    type: Constants.LOG.TYPE.EXCEPTION,
    data: {
      message: _exception,
      origin,
      time: time || new Date().getTime(),
    },
  };
}

export function clear() {
  return {
    type: Constants.LOG.CLEAR,
  };
}
