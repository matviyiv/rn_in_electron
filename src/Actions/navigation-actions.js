import Constants from '../constants';

export function navigate(screenName, payload) {
  return {
    type: Constants.NAVIGATION.NAVIGATE,
    data: { screenName, payload },
  };
}

export function goBack(payload) {
  return {
    type: Constants.NAVIGATION.GO_BACK,
    data: { payload },
  };
}
