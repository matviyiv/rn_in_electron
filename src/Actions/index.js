import * as _logActions from './log-actions';
import * as _navigationActions from './navigation-actions';

export const logActions = _logActions;
export const navigationActions = _navigationActions;
