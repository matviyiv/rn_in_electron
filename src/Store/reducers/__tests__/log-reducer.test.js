import logReducer from '../log-reducer';
import { logActions, appActions } from '../../../Actions';

describe('logReducer', () => {
  it('should return initial state if state is undefined', () => {
    expect(logReducer(undefined, {})).toMatchSnapshot();
  });

  it('should set enabled to true', () => {
    const action = logActions.enable(true);
    expect(logReducer(undefined, action)).toMatchSnapshot();
  });

  it('should set a new object in events if enabled=true', () => {
    const action = logActions.logInfo('my message', 'some origin', 12345);
    const actual = logReducer({ enabled: true }, action);
    expect(actual).toMatchSnapshot();
    expect(actual.events.length).toBe(1);
  });

  it('should not set a new object in events if enabled=false', () => {
    const action = logActions.logInfo('my message', 'some origin', 12345);
    const actual = logReducer(undefined, action);
    expect(actual.events.length).toBe(0);
  });

  it('JSON stringifies non-strings before adding to the events list', () => {
    const error = new Error('My error');
    const action = logActions.logException(error, 'some error function');
    const actual = logReducer({ enabled: true }, action);
    expect(actual.events.length).toBe(1);
    expect(JSON.parse(actual.events[0].message).message).toBe('My error');
  });

  it('should set events to empty object', () => {
    const action = logActions.clear();
    const actual = logReducer(
      {
        enabled: true,
        events: [{ foo: 'bar' }, { foo: 'bar' }, { foo: 'bar' }],
      },
      action,
    );
    expect(actual).toMatchSnapshot();
    expect(actual.events.length).toBe(0);
  });
});
