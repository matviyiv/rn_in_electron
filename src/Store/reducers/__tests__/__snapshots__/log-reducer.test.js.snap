exports[`logReducer should return initial state if state is undefined 1`] = `
Object {
  "enabled": false,
  "events": Array [],
}
`;

exports[`logReducer should set a new object in events if enabled=true 1`] = `
Object {
  "enabled": true,
  "events": Array [
    Object {
      "message": "my message",
      "origin": "some origin",
      "time": 12345,
      "type": "INFO",
    },
  ],
}
`;

exports[`logReducer should set enabled to true 1`] = `
Object {
  "enabled": true,
  "events": Array [],
}
`;

exports[`logReducer should set events to empty object 1`] = `
Object {
  "enabled": true,
  "events": Array [],
}
`;
